operator/pkg/s3.go

```diff
@@ -35,14 +37,22 @@ func (o *Operator) getBatchFromS3(proofUrl string) ([]VerificationData, error) {
 		}
 	}(resp.Body)
 
-	proof, err := io.ReadAll(resp.Body)
+	batchBytes, err := io.ReadAll(resp.Body)
 	if err != nil {
 		return nil, err
 	}
 
+	// Checks if downloaded merkle root is the same as the expected one
+	o.Logger.Infof("Verifying batch merkle tree...")
+	merkle_root_check := merkle_tree.VerifyMerkleTreeBatch(batchBytes, uint(len(batchBytes)), expectedMerkleRoot)
+	if !merkle_root_check {
+		return nil, fmt.Errorf("merkle root check failed")
+	}
```

Comment:
We shouldn't return an error, instead we should log a warning

Reply:
We also do `return fmt.Errorf("invalid proof")` on line 206 of operator.go